test/libsolidity/semanticTests/abiEncoderV2/calldata_array.sol

```diff
@@ -21,6 +21,6 @@ contract C {
 // f(uint256[][1]): 32, 32, 0 -> true
 // f(uint256[][1]): 32, 32, 1, 42 -> true
 // f(uint256[][1]): 32, 32, 8, 421, 422, 423, 424, 425, 426, 427, 428 -> true
-// gas irOptimized: 227075
```

Comment:
Would be interesting to see some of the diffs here.

Reply:
Unfortunately the output of `solc --experimental-via-ir --optimize --asm` is empty so the best I can give you is the diff of `--opcodes`:

- [opcodes-before.txt](https://github.com/ethereum/solidity/files/6387457/opcodes-before.txt)
- [opcodes-after.txt](https://github.com/ethereum/solidity/files/6387458/opcodes-after.txt)
- [opcodes-diff.txt](https://github.com/ethereum/solidity/files/6387459/opcodes-diff.txt)

Pasting the diff here makes the comment very long but has nice highlighting so let's try:
```diff
@@ -6,19 +6,18 @@
 CALLVALUE
 ISZERO
 PUSH2
-0x10
+0xF
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH2
-0x6AC
+0x5EC
 DUP1
 PUSH2
-0x21
+0x1F
 PUSH1
 0x0
 CODECOPY
@@ -26,7 +25,6 @@
 PUSH1
 0x0
 RETURN
-POP
 INVALID
 PUSH1
 0x80
@@ -37,10 +35,8 @@
 0x4
 CALLDATASIZE
 LT
-ISZERO
-ISZERO
 PUSH2
-0xCB
+0xBD
 JUMPI
 PUSH1
 0x0
@@ -54,23 +50,23 @@
 DUP2
 EQ
 PUSH2
-0x30
+0x2E
 JUMPI
 PUSH4
 0xBAF71609
 DUP2
 EQ
 PUSH2
-0x97
+0x8F
 JUMPI
 PUSH2
-0xC8
+0xBA
 JUMP
 JUMPDEST
 CALLVALUE
 ISZERO
 PUSH2
-0x3A
+0x38
 JUMPI
 DUP2
 DUP3
@@ -78,15 +74,15 @@
 JUMPDEST
 PUSH1
 0x20
+CALLDATASIZE
 PUSH1
 0x3
 NOT
-CALLDATASIZE
 ADD
 SLT
 ISZERO
 PUSH2
-0x4B
+0x49
 JUMPI
 DUP2
 DUP3
@@ -101,7 +97,7 @@
 GT
 ISZERO
 PUSH2
-0x62
+0x60
 JUMPI
 DUP3
 DUP4
@@ -115,67 +111,64 @@
 GT
 ISZERO
 PUSH2
-0x71
+0x6F
 JUMPI
 DUP3
 DUP4
 REVERT
 JUMPDEST
 PUSH2
-0x7D
+0x7B
 DUP2
 PUSH1
 0x4
 ADD
 PUSH2
-0x466
+0x3D8
 JUMP
 JUMPDEST
-SWAP1
-POP
 PUSH1
 0x40
 MLOAD
-DUP1
-PUSH2
-0x8D
-DUP4
-DUP4
-PUSH2
-0x2F7
-JUMP
+DUP2
+ISZERO
+ISZERO
+DUP2
+MSTORE
+SWAP1
+SWAP2
+POP
+PUSH1
+0x20
+DUP2
+RETURN
 JUMPDEST
 SUB
 DUP2
 RETURN
-POP
-POP
-PUSH2
-0xC8
-JUMP
 JUMPDEST
 CALLVALUE
 ISZERO
 PUSH2
-0xA1
+0x99
 JUMPI
 DUP2
 DUP3
 REVERT
 JUMPDEST
 PUSH2
-0xAA
+0xA2
 CALLDATASIZE
 PUSH2
-0x157
+0x139
 JUMP
 JUMPDEST
 POP
 POP
 PUSH2
-0xB4
+0xAC
 PUSH2
-0x5E0
+0x549
 JUMP
 JUMPDEST
 PUSH1
@@ -183,30 +176,20 @@
 MLOAD
 DUP1
 PUSH2
-0xC2
+0x8B
 DUP4
 DUP4
 PUSH2
-0x30B
+0x2B0
 JUMP
 JUMPDEST
-SUB
-DUP2
-RETURN
-POP
-POP
-JUMPDEST
 POP
 POP
 JUMPDEST
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
-PUSH2
-0x6AB
-JUMP
 JUMPDEST
 PUSH1
 0x0
@@ -216,15 +199,14 @@
 GT
 ISZERO
 PUSH2
-0xF0
+0xDC
 JUMPI
 PUSH2
-0xEF
+0xDC
 PUSH2
-0x674
+0x5CC
 JUMP
 JUMPDEST
-JUMPDEST
 DUP3
 PUSH1
 0x5
@@ -235,43 +217,38 @@
 PUSH1
 0x20
 PUSH2
-0x105
+0xF1
 DUP2
 DUP5
 ADD
 DUP4
 PUSH2
-0x435
+0x3AB
 JUMP
 JUMPDEST
-DUP2
-SWAP4
-POP
-DUP2
-DUP7
-DUP4
+DUP6
+DUP3
 MSTORE
-DUP2
-DUP4
-ADD
 SWAP1
-POP
-DUP6
 SWAP3
 POP
-DUP8
+DUP4
+SWAP1
+DUP1
 DUP5
-DUP8
 ADD
-GT
+DUP4
+DUP4
+ADD
+DUP9
+LT
 ISZERO
 PUSH2
-0x124
+0x10B
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH1
@@ -284,38 +261,33 @@
 LT
 ISZERO
 PUSH2
-0x14B
+0x12E
 JUMPI
 DUP3
 CALLDATALOAD
 DUP2
 MSTORE
-DUP2
-DUP2
-ADD
+PUSH1
+0x1
+SWAP4
 SWAP1
-POP
-DUP2
-DUP4
+SWAP4
 ADD
 SWAP3
-POP
-JUMPDEST
-PUSH1
-0x1
-DUP5
+SWAP2
+DUP2
+ADD
+SWAP2
+DUP2
 ADD
-SWAP4
-POP
 PUSH2
-0x129
+0x110
 JUMP
 JUMPDEST
 POP
 POP
 POP
 POP
-JUMPDEST
 SWAP4
 SWAP3
 POP
@@ -325,8 +297,7 @@
 JUMPDEST
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 PUSH1
 0x20
 PUSH1
@@ -337,12 +308,11 @@
 SLT
 ISZERO
 PUSH2
-0x16E
+0x14E
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH1
@@ -355,12 +325,11 @@
 GT
 ISZERO
 PUSH2
-0x188
+0x167
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 DUP5
@@ -369,15 +338,12 @@
 DUP4
 ADD
 SLT
-ISZERO
-ISZERO
 PUSH2
-0x19A
+0x176
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 DUP2
@@ -390,12 +356,11 @@
 GT
 ISZERO
 PUSH2
-0x1AD
+0x188
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 DUP6
@@ -411,12 +376,11 @@
 GT
 ISZERO
 PUSH2
-0x1C3
+0x19D
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH1
@@ -431,7 +395,6 @@
 POP
 POP
 POP
-JUMPDEST
 SWAP2
 POP
 SWAP2
@@ -447,12 +410,11 @@
 SLT
 ISZERO
 PUSH2
-0x1E8
+0x1C0
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 DUP2
@@ -463,54 +425,47 @@
 GT
 ISZERO
 PUSH2
-0x200
+0x1D7
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
-DUP1
-DUP4
+DUP3
 ADD
-SWAP1
-POP
-DUP4
 PUSH1
 0x1F
-DUP3
+DUP2
 ADD
-SLT
-ISZERO
-ISZERO
+DUP5
+SGT
 PUSH2
-0x217
+0x1E8
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 DUP1
 MLOAD
 PUSH2
-0x222
+0x1F3
 DUP2
 PUSH2
-0x341
+0x2E3
 JUMP
 JUMPDEST
 PUSH1
 0x40
 MLOAD
 PUSH2
-0x22F
+0x200
 DUP3
 DUP3
 PUSH2
-0x435
+0x3AB
 JUMP
 JUMPDEST
 DUP3
@@ -526,16 +481,15 @@
 GT
 ISZERO
 PUSH2
-0x245
+0x215
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH2
-0x256
+0x226
 DUP4
 PUSH1
 0x20
@@ -546,32 +500,28 @@
 DUP8
 ADD
 PUSH2
-0x400
+0x37B
 JUMP
 JUMPDEST
-DUP1
-SWAP5
-POP
+SWAP7
+SWAP6
 POP
 POP
 POP
 POP
-JUMPDEST
-SWAP3
-SWAP2
 POP
 POP
 JUMP
 JUMPDEST
 PUSH1
 0x20
-DUP2
-MSTORE
-DUP3
-PUSH1
-0x20
+DUP1
 DUP3
+MSTORE
+DUP2
 ADD
+DUP4
+SWAP1
 MSTORE
 PUSH1
 0x0
@@ -587,12 +537,11 @@
 GT
 ISZERO
 PUSH2
-0x285
+0x250
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 DUP4
@@ -607,122 +556,84 @@
 ADD
 CALLDATACOPY
 PUSH1
-0x40
-DUP2
-DUP5
-ADD
+0x0
+SWAP3
 ADD
-SWAP1
-POP
 PUSH1
-0x0
-DUP2
-MSTORE
-DUP1
+0x40
+ADD
 SWAP2
+DUP3
+MSTORE
 POP
-POP
-JUMPDEST
-SWAP4
 SWAP3
-POP
+SWAP2
 POP
 POP
 JUMP
 JUMPDEST
 PUSH1
-0x0
-PUSH1
 0x20
 DUP1
-DUP4
-ADD
-DUP2
-DUP5
+DUP3
 MSTORE
-DUP1
-DUP6
+DUP3
 MLOAD
-DUP1
-DUP4
+DUP3
+DUP3
+ADD
+DUP2
+SWAP1
 MSTORE
 PUSH1
-0x40
-DUP7
-ADD
+0x0
 SWAP2
-POP
-DUP4
-DUP8
+SWAP1
+DUP5
+DUP3
 ADD
-SWAP3
-POP
+SWAP1
 PUSH1
-0x0
+0x40
+DUP6
+ADD
+SWAP1
+DUP5
 JUMPDEST
 DUP2
 DUP2
 LT
 ISZERO
 PUSH2
-0x2E8
+0x2A4
 JUMPI
 DUP4
 MLOAD
 DUP4
 MSTORE
+SWAP3
 DUP5
-DUP4
 ADD
 SWAP3
-POP
-DUP5
+SWAP2
 DUP5
 ADD
-SWAP4
-POP
-JUMPDEST
+SWAP2
 PUSH1
 0x1
-DUP2
 ADD
-SWAP1
-POP
 PUSH2
-0x2C6
+0x288
 JUMP
 JUMPDEST
 POP
-POP
-DUP1
-SWAP4
+SWAP1
+SWAP7
+SWAP6
 POP
 POP
 POP
 POP
-JUMPDEST
-SWAP3
-SWAP2
-POP
-POP
-JUMP
-JUMPDEST
-PUSH1
-0x0
-PUSH1
-0x20
-DUP3
-ADD
-SWAP1
-POP
-DUP3
-ISZERO
-ISZERO
-DUP3
-MSTORE
-JUMPDEST
-SWAP3
-SWAP2
 POP
 POP
 JUMP
@@ -742,7 +653,7 @@
 ADD
 MSTORE
 PUSH2
-0x32A
+0x2CF
 DUP2
 PUSH1
 0x40
@@ -753,26 +664,23 @@
 DUP8
 ADD
 PUSH2
-0x400
+0x37B
 JUMP
 JUMPDEST
 PUSH1
-0x40
-PUSH1
 0x1F
-NOT
+ADD
 PUSH1
 0x1F
-DUP4
-ADD
+NOT
 AND
-DUP5
+SWAP2
+SWAP1
+SWAP2
 ADD
+PUSH1
+0x40
 ADD
-SWAP2
-POP
-POP
-JUMPDEST
 SWAP3
 SWAP2
 POP
@@ -787,39 +695,31 @@
 GT
 ISZERO
 PUSH2
-0x35C
+0x2FD
 JUMPI
 PUSH2
-0x35B
+0x2FD
 PUSH2
-0x674
+0x5CC
 JUMP
 JUMPDEST
-JUMPDEST
-PUSH1
-0x20
+POP
 PUSH1
 0x1F
-NOT
+ADD
 PUSH1
 0x1F
-DUP5
-ADD
+NOT
 AND
+PUSH1
+0x20
 ADD
 SWAP1
-POP
-JUMPDEST
-SWAP2
-SWAP1
-POP
 JUMP
 JUMPDEST
 DUP1
-ISZERO
-ISZERO
 PUSH2
-0x38C
+0x326
 JUMPI
 PUSH4
 0x4E487B71
@@ -841,13 +741,11 @@
 REVERT
 JUMPDEST
 POP
-JUMPDEST
 JUMP
 JUMPDEST
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 DUP3
 CALLDATALOAD
 PUSH1
@@ -859,22 +757,16 @@
 ADD
 DUP2
 SLT
-ISZERO
-ISZERO
 PUSH2
-0x3AB
+0x340
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
-DUP1
-DUP5
+DUP4
 ADD
-SWAP1
-POP
 DUP1
 CALLDATALOAD
 PUSH8
@@ -883,12 +775,11 @@
 GT
 ISZERO
 PUSH2
-0x3C8
+0x359
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH1
@@ -907,42 +798,20 @@
 SGT
 ISZERO
 PUSH2
-0x3E1
+0x371
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
-DUP2
+SWAP1
+SWAP5
+SWAP1
 SWAP4
 POP
-DUP1
-SWAP3
-POP
-POP
-POP
-JUMPDEST
-SWAP2
-POP
 SWAP2
-JUMP
-JUMPDEST
-PUSH1
-0x0
-PUSH1
-0xFF
-PUSH1
-0xF8
-SHL
-DUP3
-AND
-SWAP1
 POP
-JUMPDEST
-SWAP2
-SWAP1
 POP
 JUMP
 JUMPDEST
@@ -954,25 +823,21 @@
 LT
 ISZERO
 PUSH2
-0x41F
+0x396
 JUMPI
-DUP1
-DUP3
+DUP2
+DUP2
 ADD
 MLOAD
-DUP2
-DUP5
+DUP4
+DUP3
 ADD
 MSTORE
-JUMPDEST
 PUSH1
 0x20
-DUP2
 ADD
-SWAP1
-POP
 PUSH2
-0x403
+0x37E
 JUMP
 JUMPDEST
 DUP4
@@ -980,7 +845,7 @@
 GT
 ISZERO
 PUSH2
-0x42E
+0x3A5
 JUMPI
 PUSH1
 0x0
@@ -993,68 +858,60 @@
 POP
 POP
 POP
-JUMPDEST
 JUMP
 JUMPDEST
 PUSH1
 0x1F
-NOT
+DUP3
+ADD
 PUSH1
 0x1F
-DUP4
-ADD
+NOT
 AND
 DUP2
 ADD
-DUP2
-DUP2
-LT
 PUSH8
 0xFFFFFFFFFFFFFFFF
-DUP3
+DUP2
 GT
+DUP3
+DUP3
+LT
 OR
 ISZERO
 PUSH2
-0x45C
+0x3D1
 JUMPI
 PUSH2
-0x45B
+0x3D1
 PUSH2
-0x674
+0x5CC
 JUMP
 JUMPDEST
-JUMPDEST
-DUP1
 PUSH1
 0x40
 MSTORE
 POP
 POP
-POP
-JUMPDEST
 JUMP
 JUMPDEST
 PUSH1
 0x0
 PUSH2
-0x471
+0x3E3
 DUP3
 PUSH2
-0x390
+0x329
 JUMP
 JUMPDEST
 ADDRESS
 EXTCODESIZE
-ISZERO
-ISZERO
 PUSH2
-0x47F
+0x3EE
 JUMPI
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 REVERT
 JUMPDEST
 PUSH1
@@ -1072,17 +929,17 @@
 0x0
 DUP1
 DUP4
-DUP5
+DUP1
 PUSH2
-0x4A0
+0x40F
 DUP8
 DUP10
 PUSH1
 0x4
-DUP10
+DUP5
 ADD
 PUSH2
-0x264
+0x230
 JUMP
 JUMPDEST
 SUB
@@ -1093,56 +950,53 @@
 SWAP5
 POP
 DUP5
-ISZERO
-ISZERO
 PUSH2
-0x4B7
+0x423
 JUMPI
 PUSH2
-0x4B6
+0x423
 PUSH2
-0x69E
+0x5E2
 JUMP
 JUMPDEST
-JUMPDEST
 DUP1
 SWAP4
 POP
 DUP5
 ISZERO
 PUSH2
-0x4DD
+0x449
 JUMPI
 RETURNDATASIZE
 DUP2
 DUP5
 RETURNDATACOPY
 PUSH2
-0x4CE
+0x43A
 RETURNDATASIZE
 DUP5
 PUSH2
-0x435
+0x3AB
 JUMP
 JUMPDEST
 PUSH2
-0x4DA
+0x446
 RETURNDATASIZE
 DUP5
 ADD
 DUP5
 PUSH2
-0x1D5
+0x1AE
 JUMP
 JUMPDEST
 SWAP4
 POP
 JUMPDEST
 PUSH2
-0x4E6
+0x452
 DUP8
 PUSH2
-0x390
+0x329
 JUMP
 JUMPDEST
 SWAP6
@@ -1150,22 +1004,20 @@
 SWAP3
 POP
 PUSH2
-0x4F5
+0x461
 CALLDATASIZE
 DUP7
 DUP6
 PUSH2
-0xD5
+0xC2
 JUMP
 JUMPDEST
 SWAP5
 POP
 ADDRESS
 EXTCODESIZE
-ISZERO
-ISZERO
 PUSH2
-0x503
+0x46D
 JUMPI
 DUP1
 DUP2
@@ -1183,14 +1035,14 @@
 DUP4
 DUP5
 PUSH2
-0x51B
+0x485
 DUP9
 PUSH1
 0x4
 DUP9
 ADD
 PUSH2
-0x2A8
+0x26C
 JUMP
 JUMPDEST
 SUB
@@ -1201,46 +1053,43 @@
 SWAP5
 POP
 DUP5
-ISZERO
-ISZERO
 PUSH2
-0x532
+0x499
 JUMPI
 PUSH2
-0x531
+0x499
 PUSH2
-0x69E
+0x5E2
 JUMP
 JUMPDEST
-JUMPDEST
 DUP1
 SWAP2
 POP
 DUP5
 ISZERO
 PUSH2
-0x558
+0x4BF
 JUMPI
 RETURNDATASIZE
 DUP2
 DUP5
 RETURNDATACOPY
 PUSH2
-0x549
+0x4B0
 RETURNDATASIZE
 DUP5
 PUSH2
-0x435
+0x3AB
 JUMP
 JUMPDEST
 PUSH2
-0x555
+0x4BC
 RETURNDATASIZE
 DUP5
 ADD
 DUP5
 PUSH2
-0x1D5
+0x1AE
 JUMP
 JUMPDEST
 SWAP2
@@ -1251,135 +1100,141 @@
 SWAP5
 POP
 PUSH2
-0x568
+0x4CF
 DUP3
 MLOAD
 DUP7
 EQ
 PUSH2
-0x36F
+0x30B
 JUMP
 JUMPDEST
-DUP1
-SWAP5
-POP
+SWAP4
 POP
 JUMPDEST
-PUSH1
-0x1
-ISZERO
-PUSH2
-0x5D2
-JUMPI
 DUP3
 MLOAD
 DUP5
 LT
-ISZERO
-ISZERO
 PUSH2
-0x583
+0x4DF
 JUMPI
 PUSH2
-0x5D2
+0x53D
 JUMP
 JUMPDEST
 PUSH2
-0x595
+0x4FA
 PUSH2
-0x590
+0x4EC
 DUP6
 DUP6
 PUSH2
-0x643
+0x5A4
 JUMP
 JUMPDEST
-PUSH2
-0x68C
+MLOAD
+PUSH1
+0x1
+PUSH1
+0x1
+PUSH1
+0xF8
+SHL
+SUB
+NOT
+AND
+SWAP1
 JUMP
 JUMPDEST
 SWAP2
 POP
 PUSH2
-0x5C1
+0x52D
 PUSH2
-0x5B4
+0x51C
 PUSH2
-0x5AF
+0x50F
 PUSH2
-0x5AA
+0x4EC
 DUP8
 DUP6
 PUSH2
-0x643
-JUMP
-JUMPDEST
-PUSH2
-0x68C
+0x5A4
 JUMP
 JUMPDEST
-PUSH2
-0x3EF
+PUSH1
+0x1
+PUSH1
+0x1
+PUSH1
+0xF8
+SHL
+SUB
+NOT
+AND
+SWAP1
 JUMP
 JUMPDEST
 PUSH1
-0xFF
+0x1
+PUSH1
+0x1
 PUSH1
 0xF8
 SHL
+SUB
+NOT
 DUP5
 AND
 EQ
 PUSH2
-0x36F
+0x30B
 JUMP
 JUMPDEST
-JUMPDEST
 PUSH2
-0x5CB
+0x536
 DUP5
 PUSH2
-0x615
+0x57B
 JUMP
 JUMPDEST
 SWAP4
 POP
 PUSH2
-0x56D
+0x4D2
 JUMP
 JUMPDEST
 POP
+PUSH1
+0x1
+SWAP6
+SWAP5
 POP
 POP
 POP
-PUSH1
-0x1
-SWAP1
 POP
-JUMPDEST
-SWAP2
-SWAP1
 POP
 JUMP
 JUMPDEST
 PUSH1
 0x0
 PUSH2
-0x5EB
+0x554
 CALLDATASIZE
 PUSH2
-0x341
+0x2E3
 JUMP
 JUMPDEST
 PUSH1
 0x40
 MLOAD
 PUSH2
-0x5F8
+0x561
 DUP3
 DUP3
 PUSH2
-0x435
+0x3AB
 JUMP
 JUMPDEST
 CALLDATASIZE
@@ -1395,20 +1250,17 @@
 CALLDATACOPY
 PUSH1
 0x0
-PUSH1
-0x20
 CALLDATASIZE
-DUP4
+DUP3
 ADD
+PUSH1
+0x20
 ADD
 MSTORE
-DUP1
 SWAP3
+SWAP2
 POP
 POP
-POP
-JUMPDEST
-SWAP1
 JUMP
 JUMPDEST
 PUSH1
@@ -1420,7 +1272,7 @@
 EQ
 ISZERO
 PUSH2
-0x637
+0x59D
 JUMPI
 PUSH4
 0x4E487B71
@@ -1441,16 +1293,11 @@
 0x0
 REVERT
 JUMPDEST
+POP
 PUSH1
 0x1
-DUP3
 ADD
 SWAP1
-POP
-JUMPDEST
-SWAP2
-SWAP1
-POP
 JUMP
 JUMPDEST
 PUSH1
@@ -1459,10 +1306,8 @@
 MLOAD
 DUP4
 LT
-ISZERO
-ISZERO
 PUSH2
-0x665
+0x5C4
 JUMPI
 PUSH4
 0x4E487B71
@@ -1483,19 +1328,12 @@
 0x0
 REVERT
 JUMPDEST
+POP
+ADD
 PUSH1
 0x20
-DUP4
-DUP4
-ADD
 ADD
 SWAP1
-POP
-JUMPDEST
-SWAP3
-SWAP2
-POP
-POP
 JUMP
 JUMPDEST
 PUSH4
@@ -1517,36 +1355,12 @@
 0x0
 REVERT
 JUMPDEST
-JUMP
-JUMPDEST
-PUSH1
-0x0
-PUSH1
-0xFF
-PUSH1
-0xF8
-SHL
-DUP3
-MLOAD
-AND
-SWAP1
-POP
-JUMPDEST
-SWAP2
-SWAP1
-POP
-JUMP
-JUMPDEST
 RETURNDATASIZE
 PUSH1
 0x0
-PUSH1
-0x0
+DUP1
 RETURNDATACOPY
 RETURNDATASIZE
 PUSH1
 0x0
 REVERT
-JUMPDEST
-JUMP
-JUMPDEST
```